#167377545 API endpoint to create comment for an article

Merkll opened the pull request:
#### What does this PR do?
This pull request creates an API endpoint to comment on an existing article

#### Description of Task to be completed
- create a comment for an existing article

#### How should this be manually tested?
* clone this repo
* Run ```npm start```
* make a ```POST``` request to ```/api/v1/articles/:article/comments```
* Sample request body ``` {
  userId: '898398jn',
 articleId: '87ty47y87bv',
 content: 'nice writeup',
}```
#### Any Background Context You want to provide?
N/A

#### What are the relevant pivotal tracker stories?

[#167377545](https://pivotaltracker.com/story/show/167377545)

#### Screenshots (if appropriate)

<img width="1094" alt="Screenshot 2019-08-01 at 6 46 05 PM" src="https://user-images.githubusercontent.com/34121552/62321088-bf0ff200-b499-11e9-8352-ca01ff7db0b9.png">
<img width="1102" alt="Screenshot 2019-08-01 at 6 46 28 PM" src="https://user-images.githubusercontent.com/34121552/62321104-c9ca8700-b499-11e9-82fa-47c4f7600614.png">
<img width="1105" alt="Screenshot 2019-08-01 at 6 46 48 PM" src="https://user-images.githubusercontent.com/34121552/62321125-d5b64900-b499-11e9-83b2-2f18e578af06.png">

#### Questions:
N/A

Commit:
feat(article comment): create comments for an article

- create comments for an existing article
- create application error object

[Finishes #167377545]

successgilli commented on src/models/Users.js:
```diff
@@ -39,10 +39,10 @@ export default (sequelize, DataTypes) => {
-      foreignKey: 'userId',
-      onDelete: 'CASCADE'
-    });
+    // Users.hasMany(models.ReadStats, {
```
why did you comment this out?

successgilli reviewed (commented):
- In your pull request,  'PR' should be written in full. 'This pull request'.
- Also, correct the typo 'create' as 'creates'

Bastieno reviewed (approved):
Awesome work!

successgilli commented:
I see you made relevant changes.
Good work

Jonathan4github committed:
feat(validation): signup validation (#17)

- create validators folder
- implement signup validation function
- implement test for signup validation

[Delivers #167377536]

iverenshaguy commented on src/middlewares/asyncWrapper.js:
```diff
@@ -0,0 +1,23 @@
+  * @return {Function} - callback that execute the controller
+  */
+export default (wrappedFunction) => {
+  const currentFunction = wrappedFunction;
```
Good implementation. What's the point of reassigning it though?

Merkll replied:
None actually. just a preference.
I will remove that

iverenshaguy commented on src/middlewares/asyncWrapper.js:
```diff
@@ -0,0 +1,23 @@
+    try {
+      const { status, data, message = '' } = await currentFunction(request, response, next);
+
+      return response.status(status).json({ status: 'success', data, message });
```
You can use optional object properties here instead of sending "message" back as an empty string i.e. 
```suggestion
      return response.status(status).json({ status: 'success', data, ...(message && { message }) });
```
This way, if message is undefined. Nothing gets added to the response object. You'll also have to remove the default property on line 16 i.e.

```javascript
const { status, data, message } = await currentFunction(request, response, next);
```

iverenshaguy commented on src/middlewares/commentValidator.js:
```diff
@@ -0,0 +1,33 @@
+  const errors = request.validationErrors();
+
+  if (errors) {
+    const listOfErrors = errors.reduce((list, error) => {
```
Good job using reduce

iverenshaguy committed:
feat(article comment): update implementation

- refactor other parts of codebase to use new implementation

[Finishes 167377545]
